File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int n, int t, int m, int p) {
    string answer = "";
    
    vector<string> v = {"A", "B", "C", "D", "E", "F"};
    
    string total = "0";
    
    for(int i = 1 ; i <= t * m ; i++) {
        string tmp;
        int num = i;
        
        while(num > 0) {
            
            if(num % n >= 10){
                tmp = v[(num % n) % 10] + tmp;
            }else{
                tmp = to_string(num % n) + tmp;
            }
            
            num /= n;
        }
        
        total = total + tmp;
    }
    
    
    for(int i = p - 1 ; i < total.size() ; i += m) {
        answer.push_back(total[i]);
        if(answer.size() == t)
            break;
    }
    
    
    return answer;
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
class Solution {
    public String solution(int n, int t, int m, int p) {
        // n진법 만들기
        String answer = "0";
        
        int num = 1;
        while(answer.length() < m * t){
            answer += convertNum(num, n);
            num++;
        }
        
        // 정답 만들기
        String ret = "";
        int cnt = 0;
        while(cnt < t){
            ret += "" + answer.charAt(p - 1 + (m * cnt));
            cnt++;
        }
        
        return ret;
    }
    
    
    String convertNum(int num, int n) {
        String ret = "";
        while(num != 0){
            String val = num % n >= 10 ? convertN(num % n) :
            String.valueOf(num % n);
            ret = val + ret;
            num /= n;
        }
        return ret;
    }
    
    String convertN(int n){
        switch (n){
            case 10:
                return "A";
            case 11:
                return "B";
            case 12:
                return "C";
            case 13:
                return "D";
            case 14:
                return "E";
            case 15:
                return "F";
            default:
                return "ERR";
        }
    }
    
    
}